lib/openstax/aws/deployment_base.rb

```diff
@@ -143,6 +144,10 @@ def secrets_substitutions(&block)
         end
       end
 
+      def tag(key, value)
+        raise 'The first argument to `tag` must not be blank' if key.blank?
```

Comment:
this tests if the key is blank.  what about the value being blank?

Reply:
there aren't restrictions on the value, e.g. `""` is a legit value potentially.

we should probably have a check for `nil` tho